android-components/components/feature/search/src/test/java/mozilla/components/feature/search/telemetry/SerpTelemetryRepositoryTest.kt

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import kotlinx.coroutines.runBlocking
+import mozilla.appservices.remotesettings.RemoteSettingsRecord
+import mozilla.appservices.remotesettings.RemoteSettingsResponse
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import mozilla.components.support.test.mock
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class SerpTelemetryRepositoryTest {
+    @Mock
+    private lateinit var mockRemoteSettingsClient: RemoteSettingsClient
+
+    private lateinit var serpTelemetryRepository: SerpTelemetryRepository
+
+    @Before
+    fun setup() {
+        MockitoAnnotations.openMocks(this)
+        serpTelemetryRepository = SerpTelemetryRepository(
+            rootStorageDirectory = mock(),
+            collectionName = "testCollection",
+            serverUrl = "https://test.server",
+            bucketName = "testBucket",
```

Comment:
Still need some server url because we will get a `MalformedException : no protocol`, when we reach here in RemoteSettingsClient `private val serverHostName = URL(serverUrl).host`
The rest I've made empty